replications/internal/queue_management.go

```diff
@@ -20,21 +20,25 @@ type replicationQueue struct {
 	done    chan struct{}
 	receive chan struct{}
 	logger  *zap.Logger
+
+	writeFunc *func([]byte) error
```

Comment:
I think the `func` type is already something like an interface / a pointer under-the-hood (you can pass `nil` as a `func`), so you could drop the extra `*` here.